pam/return.go

```diff
@@ -1,64 +1,42 @@
 package main
 
-// Various signalling return messaging to PAM.
-
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
-type pamSuccess struct {
-	brokerID string
-}
-
-// String returns the string of pamSuccess.
-func (err pamSuccess) String() string {
-	return ""
-}
-
-// pamIgnore signals PAM module to return PAM_IGNORE and Quit tea.Model.
-type pamIgnore struct {
-	localBrokerID string // Only set for local broker to store it globally.
-	msg           string
-}
+import (
+	"github.com/msteinert/pam"
+)
 
-// String returns the string of pamIgnore message.
-func (err pamIgnore) String() string {
-	return err.msg
-}
-
-// pamAbort signals PAM module to return PAM_ABORT and Quit tea.Model.
-type pamAbort struct {
-	msg string
-}
-
-// String returns the string of pamAbort message.
-func (err pamAbort) String() string {
-	return err.msg
-}
+// Various signalling return messaging to PAM.
 
-// pamSystemError signals PAM module to return PAM_SYSTEM_ERROR and Quit tea.Model.
-type pamSystemError struct {
-	msg string
+type pamReturnStatus interface {
+	Status() pam.StatusError
+	Error() string
 }
 
-// String returns the string of pamSystemError message.
-func (err pamSystemError) String() string {
-	return err.msg
+// newPamStatus returns a new pamStatus.
+func newPamStatus(status pam.StatusError, err error) pamReturnStatus {
+	return pam.NewTransactionError(err, status)
```

Comment:
So... I was re-thinking about this, and one reason for having another warpping function was to ensure that what we returns implements `pamReturnStatus` that is what we expect in the pam handling functions.

Indeed this can be done at static time too... So I can still add it together with a:

```
var _ pamReturnStatus = pam.NewTransactionError(nil, pam.ErrAbort)
```

?

Reply:
This is implemented in a fixup commit, let me know if I should use it or not.

Reply:
Interfaces in Go are "automatically" implemented, so as long as the type `pam.TransactionError` has both "methods" indicated in the pamReturnStatus interface, we don't need the ```var _ pamReturnStatus = pam.NewTransactionError(nil, pam.ErrAbort)```.

Were you getting an error when using it without this "cast"?

Reply:
Sure, but I'd like to enforce it statically since in pam.go we've a type-switch where we have a `case pamReturnStatus:`. 

Indeed it may imply other failures, but this thing of implicit implementation may lead to confusion at times :)

Reply:
btw, I think we can avoid this in another way. See new change

Reply:
[Interfaces should be small, precise and define methods signatures](https://medium.com/@meeusdylan/go-interfaces-keep-them-small-f148ae200d6b). When you define and use interfaces as return values or function parameters, you are basically saying that "This value implements these methods". In the new change, you're embedding an external type that we don't really have control of. 

The first interface implementation was fine, the only problem was the unnecessary cast.

If, for some reason, the external type changes its implementation ending up no longer implementing the `pamReturnStatus` interface, it's okay (it would be a breaking change anyway), we can then refactor our code or discuss the change upstream or pin the dependency version that attends our needs.

Reply:
> let's go back to the previous implementation

Not sure which one since I proposed 3 options (sorry :-D), I pushed the previous one but let me know if I need to reiterate.

Reply:
The first one, but without the `var _ pamReturnStatus = pam.NewTransactionError(nil, pam.ErrAbort)` assertion. Did you try running the tests without it?

Reply:
Sure, that's not required by tests themselves, I was keeping it just to make this assertion valid at compile time, and since we don't have proper tests covering this side of code yet, I liked to keep it around. With the plan of removing it when some test code will actually ensure this at runtime.

Reply:
Can you add a `//TODO: {rationale about removing it when it's covered}` then? Otherwise, the chances of us forgetting about it and it staying in the codebase for quite a while are quite high :joy:

Reply:
Yeah, this was my intention. But apparently I dropped the fixup. I've added it now.